### ISSUE TITLE
Strawberry Shake should contain error messages that are in the GraphQL response

### ISSUE THREAD

**Issue by AradAral** [ChilliCream/hotchocolate#2857]
When I send a request using Strawberry Shake and the GraphQL response contains errors, Strawberry Shake doesn't properly gather the `message`s of those errors.
For example, I have a mutation field that I'm sending a request to, using Strawberry Shake, it actually returns the following response:
```json
{
  "errors": [
    {
      "message": "Mobile phone number should only contain numbers.",
      "path": [
        "sendVerificationSms"
      ],
      "extensions": {
        "code": "INVALID_INPUT",
        "invalidField": "mobilePhoneNumber"
      }
    }
  ]
}
```

Given that this is the response, you'd expect that `IOperationResult.Errors.First().Message` would be "Mobile phone number should only contain numbers.", right?! But it's instead a generic error: "Response status code does not indicate success: 500 (Internal Server Error)."
This is obviously not the expected behavior. I should be able to have access to the message that was actually in the GraphQL response.

I'm also using HC on the back-end, if that matters.

**Comment by fredericbirke** [comment]
This is because your return type is non nullable, but you return nothing, because of the error. Following the GraphQL Spec, the server will return with statuscode 500, which the current strawberryshake version cannot handle, because it calls `EnsureSuccessStatusCode` on the underlying http message, which in turn stops the strawberryshake execution flow.

We are currently reworking strawberryshake from the ground up and this will be fixed with the rewrite. We will probably release sometime in February.

**Comment by AradAral** [comment]
@fredericbirke Okay, I now understand WHY that happens, but I'm still arguing that that shouldn't happen. `IOperationResult.Errors` should always correspond to the `errors` array *in the response*, even if the HTTP response status is 500. I would say the HTTP status code should be irrelevant to this. That's what you would expect naturally.
Don't you agree?

**Comment by michaelstaib** [comment]
We agree, that is why @fredericbirke said:

> We are currently reworking strawberryshake from the ground up and this **will be fixed** with the rewrite. We will probably release sometime in February.

So, we are working on it and might have the first preview bits end of this month.

**Comment by AradAral** [comment]
@michaelstaib Got it. Thank you all.

**Comment by michaelstaib** [comment]
This is now fixed.

**Comment by AradAral** [comment]
Thank you all for all your hard work!

**Comment by Spaier** [comment]
@michaelstaib 
Issue appeared again. Strawberry Shake 13.0.5

When HotChocolate GraphQL API returns `400` error, because of missing required field -> errors are not there.
We had to work-around in server side to return `200` instead of `400` to get error messages.

https://github.com/ChilliCream/graphql-platform/blob/main/src/StrawberryShake/Client/src/Transport.Http/ResponseEnumerator.cs#L70

**Comment by michaelstaib** [comment]
its fixed in 13.1 preview 7
